docs/okeps/okep-5193-user-defined-networks.md

```diff
@@ -0,0 +1,1468 @@
+in zone 0 via br-ex rules and the CT mark restored of which network it belonged to. Finally, we can send the packet to
+the correct GR via the right patch port, by matching on the restored CT Mark. From there, OVN will handle unSNAT’ing the
+masquerade IP and forward the packet to the original pod.
+
+To support KubeVirt live migration the GR LRP will have an extra address with the configured gateway for the layer2
+subnet (to allow the gateway IP to be independent of the node where the VM is running on). After live migration succeeds,
+OVN should send a GARP for VMs to clean up its ARP tables since the gateway IP has different mac now.
+
+The live migration feature at layer 2 described here will work only with OVN interconnect (OVN IC, which is used by OCP).
+Since there is no MAC learning between zones, so we can have the same extra address on every gateway router port, basically
+implementing anycast for this SVI address.
+
+Following is a picture that illustrate all these bits with a topology
+
+![Layer 2 Egress Topology](../images/multi-homing-l2-gw.svg)
+
+##### Services
+
+When ingress service traffic enters br-ex, there are flows installed that steer service traffic towards the OVN GR. With
+additional networks, these flows will be modified to steer traffic to the correct GR-&lt;network>’s patch port.
+
+When a host process or host networked pod on a Kubernetes node initiates a connection to a service, iptables rules will
+DNAT the nodeport or loadbalancer IP into the cluster IP, and then send the traffic via br-ex where it is masqueraded
+and sent into the OVN GR. These flows can all be modified to detect the service IP and then send to the correct
+GR-&lt;network> patch port. For example, in the br-ex (breth0) bridge today we have flows that match on packets sent
+to the service CIDR (10.96.0.0/24):
+
+```
+[root@ovn-worker ~]# ovs-ofctl dump-flows breth0 table=0 | grep 10.96
+ cookie=0xdeff105, duration=22226.373s, table=0, n_packets=41, n_bytes=4598, idle_age=19399,priority=500,ip,in_port=LOCAL,nw_dst=10.96.0.0/16 actions=ct(commit,table=2,zone=64001,nat(src=169.254.169.2))
+```
+
+Packets that are destined to the service CIDR are SNAT'ed to the masquerade IP of the host (169.254.169.2) and then
+sent to the dispatch table 2:
+
+```
+[root@ovn-worker ~]# ovs-ofctl dump-flows breth0 table=2
+ cookie=0xdeff105, duration=22266.310s, table=2, n_packets=41, n_bytes=4598, actions=mod_dl_dst:02:42:ac:12:00:03,output:"patch-breth0_ov"
+```
+
+In the above flow, all packets have the dest MAC address changed to be that of the OVN GR, and then sent on the patch port
+towards the OVN GR. With multiple networks, host access to cluster IP service flows will now be modified to be on a per
+cluster IP basis. For example, if we assume two services exist on two user defined namespaces with cluster IPs 10.96.0.5
+and 10.96.0.6. The flows would look like:
+
+```
+[root@ovn-worker ~]# ovs-ofctl dump-flows breth0 table=0 | grep 10.96
```

Comment:
final design wasn't exactly like this, we do a flow per masqueradeIP IIRC that takes it to the patch ports
Would you like to preserve the initial design as is and then do updates?
OR I can update the docs for UDNs (this is still WIP PR from my side :D ) to include updates